app/src/main/java/com/gdgistanbul/attendence/ui/event_list/EventRecyclerAdapter.kt

```diff
@@ -0,0 +1,44 @@
+package com.gdgistanbul.attendence.ui.event_list
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import coil.api.load
+import com.gdgistanbul.attendence.R
+import com.gdgistanbul.attendence.extension.inflate
+import com.gdgistanbul.attendence.extension.onClick
+import com.gdgistanbul.extension.formatDateTime
+import com.gdgistanbul.model.Event
+import kotlinx.android.synthetic.main.item_event.view.*
+
+class EventRecyclerAdapter(
+    var onClickEvent: (event: Event) -> Unit = {}
+) : ListAdapter<Event, EventRecyclerAdapter.EventViewHolder>(DIFF_CALLBACK) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
+        EventViewHolder(parent.inflate(R.layout.item_event, false))
+
+    override fun onBindViewHolder(holder: EventViewHolder, position: Int) = with(holder.itemView) {
+        val event = currentList[position]
+
+        onClick { onClickEvent(event) }
```

Comment:
We could binging operation into viewHolder. May be could be like; 
`override fun onBindViewHolder(holder: EventViewHolder, position: Int) = 
 holder.bind(currentList[position])`

Reply:
Literally speaking, `ViewHolder` is responsible from "holding view" and `onBindViewHolder` is responsible from "binding view holder". If we move binding operation to the view holder, we should pass data class, listeners etc. stuff and create unnecessary couplings on view holder. I'm not sure this is the right way.

Reply:
I agree, viewHolder is responsible from holding view, but this isn't its only responsibility. I think, If view holder is holding the view, It can bind data to view. This isn't wrong. Also, some sample project of Google has this way. I'm not sure, What do you think; @oshamahue @okanaydin 

Sunflower; https://github.com/android/sunflower/blob/2da1ffa71f687e5eeeee0f9b0513782caed8d3cc/app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt

Architecture Blueprints; 
https://github.com/android/architecture-samples/blob/89becf6025b9caedcbc669138379f21ada02a659/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksAdapter.kt

Reply:
With this method, an onClick is created for each item in every bind transaction. In fact, it is enough to create a single onClick
        init {
            itemView.setOnClickListener {
                onClickEvent(event)
            }
        }
Sample;
https://github.com/android/sunflower/blob/master/app/src/main/java/com/google/samples/apps/sunflower/adapters/GardenPlantingAdapter.kt

Reply:
@mertceyhan I'm working on a DSL currently. I think, everyone will be much more happy now.
@perihanimamoglu Good detection. I'll consider this.